### ISSUE TITLE
The options.env property on VS Code tasks seems to be ignored by Docker tasks

### ISSUE THREAD

**Issue by Xitric** [microsoft/vscode-docker#3437]
In our development environment, we are migrating to docker compose for spinning up our products locally. The configuration of such an environment consists of some static arguments which we have checked into git, as well as some user-specific arguments. We would like to avoid asking of our developers to create their own `.env` files that are not tracked by git, but rather leverage the power of inputs in VS Code tasks.

Let us say that a given compose environment requires knowledge of the current user's email address to bootstrap an admin account. The `docker-compose.yaml` file may contain references to three environment variables like this:

```yaml
product1:
    image: ${CONTAINER_REGISTRY:-company}/product1:${TAG:-latest}
    build:
      context: .
      dockerfile: Dockerfile.Product1
    environment:
      - ADMIN_USER=${USER_EMAIL:?Admin user email is missing or empty}
```

For `${CONTAINER_REGISTRY}` and `${TAG}`, we can place these in an `.env` file alongside the `docker-compose.yaml` file, since they will typically not be edited except temporarily during testing. As for `${USER_EMAIL}`, however, this will be different for every developer. Thus, we have so far been using a command such as the following to spin up the environment:

```bash
USER_EMAIL=my@email.com docker compose up -d
```

Recently, we have begun exploring VS Code tasks to simplify keeping track of the many (sometimes complex) compose commands we use, and the following does the trick:

```json
{
    "label": "Compose up: Product 1",
    "type": "shell",
    "command": "docker compose up -d",
    "options": {
        "env": {
            "USER_EMAIL": "${input:userEmail}"
        }
    }
}
```

However, when trying to do the same with the dedicated `docker-compose` task type, the environment variable is not resolving in the `docker-compose.yaml` file:

```json
{
    "label": "Compose up: Product 1",
    "type": "docker-compose",
    "dockerCompose": {
        "up": {
            "detached": true,
            "build": true
        },
        "files": [
            "${workspaceFolder}/docker-compose.yaml"
        ],
        "envFile": "${workspaceFolder}/.env"
    },
    "options": {
        "env": {
            "USER_EMAIL": "${input:userEmail}"
        }
    }
}
```

Running this gives:

```none
invalid interpolation format for services.product1.environment.[]: "required variable USER_EMAIL is missing a value: Admin user email is missing or empty".
```

**Comment by bwateratmsft** [comment]
Thanks for the detailed report! We should get this fixed. I don't expect it will be _too_ difficult. I'd like to also fix it for `docker-build` (although `buildArgs` is kinda similar) and `docker-run` tasks.

**Comment by bwateratmsft** [comment]
@alexr00 the `options` field on the task is not coming through with the `resolvedDefinition` in the `CustomExecution`. Do you know why?

**Comment by karolz-ms** [comment]
I will let @alexr00 chime in, but I noticed that, unlike `ProcessExecution` and `ShellExecution`, `CustomExecution` class definition does not have `options` property...

**Comment by alexr00** [comment]
The `TaskDefinition` that's passed into the `CustomExecution` callback is only composed of the properties that are part of that task's definition, i.e., the properties that are listed in the providing extensions `taskDefinitions` in it's package.json.  Since it's not part of the definition of `docker-compose`, it doesn't get passed.  You might be able to get `options` if you add it to the definition of `docker-compose` in your package.json. 

Let me know if that works or has other issues and I'll see if there's any quick fix that can be done.

**Comment by bwateratmsft** [comment]
I'll give that a try, thanks @alexr00!

**Comment by bwateratmsft** [comment]
This fix has now been released in version 1.21.0 of the Docker extension.
